### PR TITLE
Introduce hierarchical metadata structures

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request introduces hierarchical metadata structures as discussed during dev call.

A registry of FITS header exporters in introduced to follow on the ideas developed in #4867 and during the dev call.

**Dear reviewer**

### CODE REVIEW

**Comment by adonath** [comment]
@registerrier I'd propose that we merge this PR fast and I rebase and continue with #4750...

**Comment by registerrier** [comment]
The `obs_info` argument on init is now deprecated and for now recreated from the `meta` with `meta.to_header()`. It does not contain exactly the same members as the full events list meta. Some tests were adapted.

The remaining issue is `obs_id` which is present both on the `meta` and as an attribute. Removing it from the main class, requires creating a `meta` but not filling it since we have to wait for the fits file to be opened. This might be solved later.

File: gammapy/data/observations.py [hunk 1]
```diff
@@ -331,19 +337,15 @@ def observation_dead_time_fraction(self):
         """
         return self.meta.deadtime_fraction
 
-    @lazyproperty
+    #    @deprecated("v1.2", alternative="meta")
```
Comment (registerrier) [inline gammapy/data/observations.py:340]:
```suggestion
```

File: gammapy/data/observations.py [hunk 2]
```diff
@@ -550,7 +552,7 @@ def read(cls, event_file, irf_file=None):
         return cls(
             events=events,
             gti=gti,
-            obs_info=obs_info,
+            #            obs_info=obs_info,
```
Comment (registerrier) [inline gammapy/data/observations.py:555]:
```suggestion
```

File: tox.ini [hunk 3]
```diff
@@ -77,7 +77,7 @@ deps =
     oldestdeps: astropy==5.0.*
     oldestdeps: click==7.0.*
     oldestdeps: regions==0.5.*
-    oldestdeps: pydantic==1.4.*
+    oldestdeps: pydantic==1.5.*
```
Comment (registerrier) [inline tox.ini:80]:
```suggestion
    oldestdeps: pydantic==1.4.*
```

**Comment by registerrier** [comment]
Merging now.

**Pull request merged**